File: java/Sixth/src/test/q5/Person.java
package test.q5;

public class Person {

    public String name;
    public int money;

    public Person(String name, int money) {

        this.name = name;
        this.money = money;
    }

    public void buyStarCoffee(Starbucks starCoffee, int money) {

        String message = starCoffee.brewing(4000);

        if (message != null) {
            this.money -= money;
            System.out.println(name + "씨는 " + money + "원을 내고 " + message);
        }
    }

    public void buyBeanCoffee(Coffeebean beanCoffee, int money) {

        String message = beanCoffee.brewing(4500);

        if (message != null) {
            this.money -= money;
            System.out.println(name + "씨는 " + money + "원을 내고 " + message);
        }
    }
     
}

File: python/p_oop2/chap05/excercise/coffee_module/person.py
class Person:
    def __init__(self, name, money):
        self.name = name
        self.money = money

    def buy_star_coffee(self, s_coffee, money):
        message = s_coffee.brewing(money)
        if message != None:
            self.money -= money
            print(self.name + "님이 " + str(money) + "으로 " + message)

    def buy_bean_coffee(self, b_coffee, money):
        message = b_coffee.brewing(money)
        if message != None:
            self.money -= money
            print(self.name + "님이 " + str(money) + "으로 " + message)
